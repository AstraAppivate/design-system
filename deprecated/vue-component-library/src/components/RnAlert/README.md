---
title: Alert
description: Alerts provide an easy to deliver messages to the user.
---

```html
  <div class="rn-alert">
    <p class="rn-alert__message">This is an example alert message</p>
  </div>
```

## Title

An optional title can be added to the alert.

```html
  <div class="rn-alert">
    <h4 class="rn-alert__title">Alert Title</h4>
    <p class="rn-alert__message">This is an example alert message</p>
  </div>
```

# Event Alerts

Event alerts are small indicators that provide feedback to the user.

```html
  <div class="rn-alert--event">
    This is an Event Alert
  </div>
```

#### Props

Prop Name | Value | Required
--------- | ----- | --------
**title** | `{String}` | false
**state** | `default` `success` `info` `danger` | false