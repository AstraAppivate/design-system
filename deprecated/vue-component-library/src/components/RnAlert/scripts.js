import RnButton from '../RnButton/index.vue'
import RnCard from '../RnCard/index.vue'

export default {
  name: 'RnAlert',
  components: {
    RnCard,
    RnButton,
  },
  data: () => ({
    isOpen: false,
  }),
  props: {
    title: String,
    state: {
      type: String,
      default: 'neutral',
    },
    hideClose: {
      type: Boolean,
      default: false,
    },
    hideAction: {
      type: Boolean,
      default: false,
    },
    actionButtonText: {
      type: String,
      default: 'OK',
    },
    error: Boolean,
  },
}
