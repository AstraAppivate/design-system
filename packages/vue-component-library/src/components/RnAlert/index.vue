<template lang="html">
  <div v-if="open" :class="classes">
    <h4 v-if="title" class="title">{{ title }}</h4>
    <p class="message">
      <slot></slot>
    </p>
    <button type="button" class="close" @click.prevent="close">
      <svg xmlns="http://www.w3.org/2000/svg" width="12" height="12">
        <path
          fill="currentColor"
          fillRule="evenod"
          d="M8.09 6l3.48 3.48a1.48 1.48 0 1 1-2.09 2.09L6 8.09l-3.48 3.48A1.48 1.48 0 0 1 .43 9.48L3.91 6 .43 2.52A1.48 1.48 0 1 1 2.52.43L6 3.91 9.48.43a1.48 1.48 0 0 1 2.09 2.09L8.09 6z"
        />
      </svg>
    </button>
  </div>
</template>

<script src="./scripts.js"></script>
