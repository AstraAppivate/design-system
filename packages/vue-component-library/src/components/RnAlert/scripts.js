import RnCard from '../RnCard/index.vue'

export default {
  name: 'RnAlert',

  components: {
    RnCard,
  },

  props: {
    title: {
      type: String,
    },
    state: {
      type: String,
      default: 'default',
    },
  },

  data: () => ({
    open: true,
  }),

  computed: {
    classes() {
      return `rn-alert ${this.state}`
    },
  },

  methods: {
    close() {
      this.open = false
      this.$emit('close')
    },
  },
}
