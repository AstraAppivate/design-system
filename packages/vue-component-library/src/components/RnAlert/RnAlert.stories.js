import { storiesOf } from '@storybook/vue'

import RnAlert from './index.vue'

storiesOf('Alerts', module)
  .add('Default', () => ({
    components: { RnAlert },
    template: '<rn-alert>This is an example alert message </rn-alert>',
  }))
  .add('Default with title', () => ({
    components: { RnAlert },
    template:
      '<rn-alert title="Alert message with title">This is an example alert message</rn-alert>',
  }))
  .add('Success', () => ({
    components: { RnAlert },
    template:
      '<rn-alert title="Success Title" state="success">This is an example alert message</rn-alert>',
  }))
  .add('Info', () => ({
    components: { RnAlert },
    template:
      '<rn-alert title="Info Title" state="info">This is an example alert message</rn-alert>',
  }))
  .add('Danger', () => ({
    components: { RnAlert },
    template:
      '<rn-alert state="danger">This is an example alert message</rn-alert>',
  }))
