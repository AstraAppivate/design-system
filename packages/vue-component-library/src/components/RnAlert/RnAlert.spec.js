import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import RnAlert from '@/src/components/RnAlert/index.vue'

describe('RnAlert.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnAlert, {
      propsData: {
        text: 'foo',
      },
    })
  })

  it('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
