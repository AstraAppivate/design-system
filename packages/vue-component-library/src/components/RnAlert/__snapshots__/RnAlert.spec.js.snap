// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnAlert.vue snapshot: has same HTML structure 1`] = `
<div text="foo" class="rn-alert default">
  <!---->
  <p class="message"></p> <button type="button" class="close"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12">
      <path fill="currentColor" fillRule="evenod" d="M8.09 6l3.48 3.48a1.48 1.48 0 1 1-2.09 2.09L6 8.09l-3.48 3.48A1.48 1.48 0 0 1 .43 9.48L3.91 6 .43 2.52A1.48 1.48 0 1 1 2.52.43L6 3.91 9.48.43a1.48 1.48 0 0 1 2.09 2.09L8.09 6z"></path>
    </svg></button>
</div>
`;
